Flaky test: TestRemovePath, error is not considered retriable

## Flaky Test

* **Test Name:** internal/pkg/agent/install TestRemovePath

* **Link:** [uninstall_windows_test.go](https://github.com/elastic/elastic-agent/blob/0f3229248d82145a49005204ebff2d8a1cb864fc/internal/pkg/agent/install/uninstall_windows_test.go#L31)

* **Branch:** 
 - main
 - others

* **Artifact Link:** 
  - [Jenkins](https://fleet-ci.elastic.co/blue/organizations/jenkins/elastic-agent%2Felastic-agent-mbp/detail/main/1298/pipeline/2133/)

* **Notes:** 
 - the issue seems to be on how the error is considered retriable 
 - this failure happened on Windows 8

### Stack Trace

```
=== RUN   TestRemovePath
    uninstall_windows_test.go:54: 
        	Error Trace:	C:/Users/jenkins/workspace/tic-agent_elastic-agent-mbp_main/src/github.com/elastic/elastic-agent/internal/pkg/agent/install/uninstall_windows_test.go:54
        	Error:      	Received unexpected error:
        	            	could not remove "C:\\Users\\jenkins\\AppData\\Local\\Temp\\TestRemovePath2573796789\\001\\subdir", unretriable error: failed to open handle for "C:\\Users\\jenkins\\AppData\\Local\\Temp\\TestRemovePath2573796789\\001\\subdir\\main.exe": The process cannot access the file because it is being used by another process.
        	Test:       	TestRemovePath
--- FAIL: TestRemovePath (0.62s)
```

--- a/changelog/fragments/1693319952-Change-error-handling-o-Windows-when-uninstalling-the-Elastic-Agent.yaml
+++ b/changelog/fragments/1693319952-Change-error-handling-o-Windows-when-uninstalling-the-Elastic-Agent.yaml
@@ -1,0 +1,32 @@
+# Kind can be one of:
+# - breaking-change: a change to previously-documented behavior
+# - deprecation: functionality that is being removed in a later release
+# - bug-fix: fixes a problem in a previous version
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: bug-fix
+
+# Change summary; a 80ish characters long description of the change.
+summary: Correctly identify retryable errors when attempting to uninstall on Windows
+
+# Long description; in case the summary is not enough to describe the change
+# this field accommodate a description without length limits.
+# NOTE: This field will be rendered only for breaking-change and known-issue kinds at the moment.
+#description:
+
+# Affected component; a word indicating the component this changeset affects.
+component: Uninstaller
+
+# PR URL; optional; the PR number that added the changeset.
+# If not present is automatically filled by the tooling finding the PR where this changelog fragment has been added.
+# NOTE: the tooling supports backports, so it's able to fill the original PR number instead of the backport PR number.
+# Please provide it if you are adding a fragment for a different PR.
+#pr: https://github.com/owner/repo/1234
+
+# Issue URL; optional; the GitHub issue related to this changeset (either closes or is part of).
+# If not present is automatically filled by the tooling with the issue linked to the PR number.
+#issue: https://github.com/owner/repo/1234

--- a/internal/pkg/agent/install/uninstall.go
+++ b/internal/pkg/agent/install/uninstall.go
@@ -134,8 +134,8 @@ func removePath(path string) error {
 		return nil
 	}
 
-	if isBlockingOnExe(err) {
-		// try to remove the blocking exe
+	if isAccessDeniedError(err) {
+		// might be blocking on exe, then try to remove the blocking exe
 		err = removeBlockingExe(err)
 	}
 

--- a/internal/pkg/agent/install/uninstall_unix.go
+++ b/internal/pkg/agent/install/uninstall_unix.go
@@ -6,7 +6,7 @@
 
 package install
 
-func isBlockingOnExe(_ error) bool {
+func isAccessDeniedError(_ error) bool {
 	return false
 }
 

--- a/internal/pkg/agent/install/uninstall_windows.go
+++ b/internal/pkg/agent/install/uninstall_windows.go
@@ -16,26 +16,31 @@ import (
 	"golang.org/x/sys/windows"
 )
 
-func isBlockingOnExe(err error) bool {
+func isAccessDeniedError(err error) bool {
 	if err == nil {
 		return false
 	}
-	path, errno := getPathFromError(err)
-	if path == "" {
-		return false
+
+	var errno syscall.Errno
+	if errors.As(err, &errno) {
+		return errno == syscall.ERROR_ACCESS_DENIED
 	}
-	return errno == syscall.ERROR_ACCESS_DENIED
+
+	return false
 }
 
 func isRetryableError(err error) bool {
 	if err == nil {
 		return false
 	}
-	path, errno := getPathFromError(err)
-	if path == "" {
-		return false
+
+	var errno syscall.Errno
+	if errors.As(err, &errno) {
+		return errno == syscall.ERROR_ACCESS_DENIED ||
+			errno == windows.ERROR_SHARING_VIOLATION
 	}
-	return errno == syscall.ERROR_ACCESS_DENIED || errno == windows.ERROR_SHARING_VIOLATION
+
+	return false
 }
 
 func removeBlockingExe(blockingErr error) error {
@@ -72,14 +77,20 @@ func removeBlockingExe(blockingErr error) error {
 	return nil
 }
 
-func getPathFromError(blockingErr error) (string, syscall.Errno) {
+func getPathFromError(err error) (string, syscall.Errno) {
 	var perr *fs.PathError
-	if errors.As(blockingErr, &perr) {
+	if errors.As(err, &perr) {
 		var errno syscall.Errno
 		if errors.As(perr.Err, &errno) {
 			return perr.Path, errno
 		}
 	}
+
+	var errno syscall.Errno
+	if errors.As(err, &errno) {
+		return "", errno
+	}
+
 	return "", 0
 }
 

